Put  IgProf back in. Remove running profiling on IB only.

gartung opened the pull request.

cmsbuild commented:
A new Pull Request was created by  @gartung for branch master.

@cmsbuild, @iarspider, @smuzaffar can you please review it and eventually sign? Thanks.
@antoniovilela, @mandrenguyen, @rappoccio, @sextonkennedy you are the release manager for this.
cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

cmsbuild commented:
cms-bot internal usage<!-- bot cache: {"commits":{"39bb0cccbbc7b4a745cb7228611b0ec2a14c0d4a":{"files":["pr_testing/run-pr-profiling.sh"],"squashed":false,"time":1743454617}},"emoji":{},"last_seen_sha":"39bb0cccbbc7b4a745cb7228611b0ec2a14c0d4a","signatures":{}} -->

cmsbuild commented:
Pull request #2467 was updated.

mandrenguyen commented:
+1

makortel commented:
@mandrenguyen I'm wondering why orp signed this PR when it didn't have the `fully-signed` label yet?

gartung commented:
please test with https://github.com/cms-sw/cmsdist/pull/9759

gartung commented:
@smuzaffar  @iarspider This removes running the profiling with IB only. That is causing the long job times for pr-profiling.

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8e3fe1/45340/summary.html
**COMMIT**: ce0c0826ceb56750f6def7c0f64db433d1225907
**CMSSW**: CMSSW_15_1_X_2025-04-02-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cms-bot/2467/45340/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [added 18 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-04-02-1100+8e3fe1/67804/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 0 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-04-02-1100+8e3fe1/67804/validateJR.html)
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 3913985
 - DQMHistoTests: [Total failures: 18](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-04-02-1100+8e3fe1/67804/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3913947
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 49 files compared)
 - Checked 215 log files, 184 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-04-02-1100+8e3fe1/67804/triggerResults): no differences found

gartung commented:
enable profiling

gartung commented:
please test with https://github.com/cms-sw/cmsdist/pull/9759

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8e3fe1/45347/summary.html
**COMMIT**: ce0c0826ceb56750f6def7c0f64db433d1225907
**CMSSW**: CMSSW_15_1_X_2025-04-02-1100/el8_amd64_gcc12
**Additional Tests**: PROFILING
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cms-bot/2467/45347/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [added 18 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-04-02-1100+8e3fe1/67812/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 7 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-04-02-1100+8e3fe1/67812/validateJR.html)
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 3913985
 - DQMHistoTests: [Total failures: 78](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-04-02-1100+8e3fe1/67812/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3913887
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 49 files compared)
 - Checked 215 log files, 184 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-04-02-1100+8e3fe1/67812/triggerResults): no differences found

smuzaffar commented:
> This removes running the profiling with IB only

@gartung , what do you mean by that?

gartung commented:
I added running with the IB alone then with the IB+PR for comparisons. Since it is taking too long, I changed it back to running the IB+PR only and downloading the results from the IB.

gartung commented on pr_testing/run-pr-profiling.sh:
```diff
@@ -27,35 +27,25 @@ for PROFILING_WORKFLOW in $WORKFLOWS;do
-	  cp -v $f $WORKSPACE/$CMSSW_VERSION/$PROFILING_WORKFLOW/$CMSSW_VERSION-$b || true
-  done
-  $WORKSPACE/profiling/Gen_tool/runall.sh $CMSSW_VERSION || true
+  unset RUNALLSTEPS
```
@smuzaffar Note than `runall.sh` was called twice in the old version.

smuzaffar commented:
ah ok, @gartung is it then ready to go in?

gartung commented:
Yes

smuzaffar commented:
+externals

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will be automatically merged.
**Notice** This PR was tested with additional Pull Request(s), please also merge them if necessary: cms-sw/cmsdist#9759
